### ISSUE TITLE
Support inherit and expend types

### ISSUE THREAD

**Issue by sergey-tihon** [fsprojects/SwaggerProvider#72]
### Schema 

https://api.apis.guru/v2/specs/azure.com/arm-storage/2016-12-01/swagger.json

### Sample
```json
"StorageAccount": {
      "allOf": [
        {
          "$ref": "#/definitions/Resource"
        }
      ],
      "description": "The storage account.",
      "properties": {
        "kind": {
          "description": "Gets the Kind.",
          "enum": [
            "Storage",
            "BlobStorage"
          ],
          "readOnly": true,
          "type": "string",
          "x-ms-enum": {
            "modelAsString": false,
            "name": "Kind"
          }
        },
        "properties": {
          "$ref": "#/definitions/StorageAccountProperties",
          "x-ms-client-flatten": true
        },
        "sku": {
          "$ref": "#/definitions/Sku",
          "description": "Gets the SKU.",
          "readOnly": true
        }
      }
    },
```

`StorageAccount` should have all properties of `Resource` type // cc @stuartjdavies
 

### Spec

Composition and Inheritance (Polymorphism)

Swagger allows combining and extending model definitions using the allOf property of JSON Schema, in effect offering model composition. allOf takes in an array of object definitions that are validated independently but together compose a single object.

While composition offers model extensibility, it does not imply a hierarchy between the models. To support polymorphism, Swagger adds the support of the discriminator field. When used, the discriminator will be the name of the property used to decide which schema definition is used to validate the structure of the model. As such, the discriminator field MUST be a required field. The value of the chosen property has to be the friendly name given to the model under the definitions property. As such, inline schema definitions, which do not have a given id, cannot be used in polymorphism.

**Comment by sergey-tihon** [comment]
Fix is on the way
<img width="1168" alt="allof-props" src="https://cloud.githubusercontent.com/assets/1197905/25010277/4e020610-2072-11e7-8622-a549f7c8b1bf.png">

**Comment by sergey-tihon** [comment]
Released in [v0.6.0](https://www.nuget.org/packages/SwaggerProvider/0.6.0)
